docs: single-page RTFD outline

diegodelemos opened the pull request:
* Addresses reanahub/reana#335.

mvidalgarcia commented:
I get this, any clue?
```console
(r-j-c) ➜  reana-job-controller git:(r-335/restructuring-docs) ✗ sphinx-build -qnN docs docs/_build/

Extension error:
Could not import extension sphinxcontrib.httpdomain (exception: No module named 'sphinxcontrib.httpdomain')
(r-j-c) ➜  reana-job-controller git:(r-335/restructuring-docs) ✗ pip show sphinxcontrib-httpdomain
Name: sphinxcontrib-httpdomain
Version: 1.7.0
Summary: Sphinx domain for documenting HTTP APIs
Home-page: https://github.com/sphinx-contrib/httpdomain
Author: Hong Minhee
Author-email: hong.minhee@gmail.com
License: BSD
Location: /Users/marco/.virtualenvs/r-j-c/lib/python3.6/site-packages
Requires: Sphinx, six
Required-by: sphinxcontrib-openapi
```

diegodelemos commented:
Did you successfully install RJC? if you did this package comes in the `setup.py` so it should be installed. Note that RJC cannot be installed on macOS, one has to either:
- Run inside Docker (see [`run-tests.sh`](https://github.com/reanahub/reana-job-controller/blob/069fbb3964c04e392c74405e785487df304915af/run-tests.sh#L89))
- Comment out [HTCondor package from `setup.py`](https://github.com/reanahub/reana-job-controller/blob/069fbb3964c04e392c74405e785487df304915af/setup.py#L55) (there is not such a package for macOS)

mvidalgarcia commented:
Yes I commented out `htcondor`.

However get this during installation:
```console
ERROR: sphinxcontrib-openapi 0.7.0 has requirement sphinx>=2.0, but you'll have sphinx 1.5.6 which is incompatible.
```

If I remove the [upper boundary of Sphinx in setup.py](https://github.com/reanahub/reana-job-controller/blob/master/setup.py#L28) I don't get that error anymore but this one:

```console
Warning, treated as error:
/Users/marco/code/reanahub/reana-job-controller/reana_job_controller/job_manager.py:docstring of reana_job_controller.job_manager:1:duplicate object description of reana_job_controller.job_manager, other instance in developerguide, use :noindex: for one of them
```
I think you forgot to remove `developerguide.rst`.

diegodelemos commented:
> ```
> (r-j-c) ➜  reana-job-controller git:(r-335/restructuring-docs) ✗ sphinx-build -qnN docs docs/_build/
> ...
> Extension error:
> Could not import extension sphinxcontrib.httpdomain (exception: No module named 'sphinxcontrib.httpdomain')
>```

I can reproduce this error by using `sphinx-build` from outside the virtualenv, you might have used `sphinx-build` from a different Python installation:
```console
$ which sphinx-build
/Library/Frameworks/Python.framework/Versions/3.8/bin/sphinx-build
```

> If I remove the upper boundary of Sphinx in setup.py I don't get that error anymore but this one:

Addressed [as in RS](https://github.com/reanahub/reana-server/blob/231e0d56626087962391c380e3446a807bc2a927/setup.py#L31), I've added an upper limit to `sphinx-openapi`.

> I think you forgot to remove developerguide.rst.

Amended, thanks!
